optimum/intel/neural_compressor/utils.py

```diff
@@ -109,3 +114,240 @@ def load_quantized_model(checkpoint_dir_or_file: str, model: torch.nn.Module, **
         )
 
     return load(checkpoint_dir_or_file, model, **kwargs)
+
+
+mapping = {
+    "CausalLMOutputWithCrossAttentions": CausalLMOutputWithCrossAttentions,
+    "CausalLMOutput": CausalLMOutput,
+    "XLNetLMHeadModelOutput": XLNetLMHeadModelOutput,
+}
+
+
+class AutoModelForGeneration(_BaseAutoModelClass):
+    _model_mapping = MODEL_FOR_CAUSAL_LM_MAPPING
+
+    # def forward(self, input_ids=None, past_key_values=None, attention_mask=None, **kwargs):
+    def __call__(self, *args, **kwargs):
+        past_key_values = getattr(kwargs, "past_key_values", None)
+        mems = getattr(kwargs, "mems", None)
+        if hasattr(self, "traced_model") and (past_key_values is not None or mems is not None):
+            inputs = []
+            kwargs.pop("position_ids", None)
```

Comment:
In case we use the generated dummy inputs from the ONNXConfig and reorder them according to the model forward signature (as suggested in an comment), we should keep the `position_ids` if given

Reply:
yes,  if we fixed dummy_inputs issue, I will remove this limitation.

Reply:
Removed